src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java

```diff
@@ -0,0 +1,461 @@
+    @Test
+    void shouldThrowExceptionWhenAccountThrowException() throws IOException {
+        when(accountsBuilderMock.buildAccount()).thenThrow(IOException.class);
+
+        assertThrows(IOException.class,
+            () -> filingService.generateAccountFiling(transaction, companyAccountEntity));
+    }
+
+    @DisplayName("Tests exception being thrown when incorrect json format")
+    @Test
+    void shouldThrownJsonProcessingException() throws IOException {
+        when(accountsBuilderMock.buildAccount()).thenReturn(getSmallFullAccount());
+
+        when(objectMapperMock.writeValueAsString(any(Object.class)))
+            .thenThrow(JsonProcessingException.class);
+
+        assertThrows(JsonProcessingException.class,
+            () -> filingService.generateAccountFiling(transaction, companyAccountEntity));
+    }
+
+    @DisplayName("Tests the filing not generated when ixbrl location has not been set, ixbrl not generated")
+    @Test
+    void shouldNotGenerateFilingAsIxbrlLocationNotSet() throws IOException {
+        when(accountsBuilderMock.buildAccount()).thenReturn(getSmallFullAccount());
+
+        when(objectMapperMock.writeValueAsString(any(Object.class))).thenReturn(smallFullJson);
+
+        when(environmentReaderMock.getMandatoryString(anyString()))
+            .thenReturn("http://localhost:4082")
+            .thenReturn("apiKeyForTesting")
+            .thenReturn("dev-pdf-bucket/chs-dev");
+
+        Filing filing = filingService.generateAccountFiling(transaction, companyAccountEntity);
+
+        verifyObjectMapperNumOfCalls();
+        verifyIxbrlGeneratorNumOfCalls();
+        verifyEnvironmentReaderNumOfCalls(3,
+            DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR,
+            API_KEY_ENV_VAR,
+            DOCUMENT_BUCKET_NAME_ENV_VAR);
+
+        assertNull(filing);
+    }
+
+    @DisplayName("Tests exception being thrown when the document render service unavailable")
+    @Test
+    void shouldThrowExceptionDocumentGeneratorIsUnavailable() throws IOException {
+        when(accountsBuilderMock.buildAccount()).thenReturn(getSmallFullAccount());
+
+        when(objectMapperMock.writeValueAsString(any(Object.class))).thenReturn(smallFullJson);
+
+        when(ixbrlGeneratorMock.generateIXBRL(any(DocumentGeneratorConnection.class)))
+            .thenThrow(new ConnectException());
+
+        assertThrows(ConnectException.class,
+            () -> filingService.generateAccountFiling(transaction, companyAccountEntity));
+    }
+
+    /**
+     * Check environmentReaderMock is called the passed-in number of times and the argument's values
+     * of these calls matches the passed-in args.
+     *
+     * @param numOfTimes The number of times environment reader is be called.
+     * @param args the values the environment reader is called with.
+     */
+    private void verifyEnvironmentReaderNumOfCalls(int numOfTimes, String... args) {
+
+        verify(environmentReaderMock, times(numOfTimes))
```

Comment:
Instead of needing the int passed in you could simplify to:
```java
private void verifyEnvironmentReaderNumOfCalls(String... args) {
   verify(environmentReaderMock, times(args.length))
```

Reply:
good shout. 💯

Reply:
2f26d58